File: lab/python/palindromos/palindromos2.py
def palindromo1(texto):
    # normaliza: tudo minúsculo, sem espaços
    normalizado = texto.lower().replace(" ", "")
    return "SIM" if normalizado == normalizado[::-1] else "NAO"


def palindromo2(texto):
    texto = texto.lower().replace(" ", "")
    indice_final = len(texto) - 1

    inverso = ""

    for i in range(indice_final, -1, -1):
        inverso += texto[i]

    return "SIM" if inverso == texto else "NAO"


def palindromo_numerico(numero):
    original = numero
    inverso = 0
    while numero > 0:
        resto = numero % 10
        inverso = inverso * 10 + resto
        numero //= 10

    return "SIM" if inverso == original else "NAO"


print(palindromo1("asa"))
print(palindromo2("radar"))
print(palindromo_numerico(121))

File: lab/javascript/palindromos/palindromos2.js
function palindromo1(texto){
    let normalizado = String(texto).toLowerCase().replace(/\s+/g, "");
    return (normalizado === normalizado.split("").reverse().join("")) ? "SIM":"NAO"; 
}

function palindromo2(texto){
    texto = String(texto).toLowerCase().replace(/\s+/g, "");
    let indice_final = texto.length - 1;

    let inverso = "";

    for (let i=indice_final; i > -1; i--){
        inverso += texto[i];
    }

    return (inverso === texto) ? "SIM":"NAO";
}

function palindromo_numerico(numero){
    let original = numero;
    let inverso = 0;

    while (numero > 0){
        let resto = numero % 10;
        inverso = inverso * 10 + resto;
        numero = Math.floor(numero / 10);  // divisão inteira
    }

    return (inverso === original) ? "SIM" : "NAO";
}


console.log(palindromo1("asa"));
console.log(palindromo2("radar"));
console.log(palindromo_numerico(121));
